packages/graphback-core/src/utils/convertType.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * Helper function to convert a value to another type
+ * 
+ * @param {any} value - Value to convert
+ * @param {string} typeOf - convert value to this type
+ */
+export function convertType(value: any, typeOf: "string" | "number" | "bigint" | "boolean" | "symbol" | "undefined" | "object" | "function"): string | number | boolean | BigInt {
+  if (!value) {
+    return undefined;
+  }
+  
+  switch (typeOf) {
+    case 'string':
+      return String(value);
+    case 'number':
+      return Number(value);
+    case 'bigint':
+      return BigInt(value);
+    case 'boolean':
+      return Boolean(value);
+    case 'object':
+      if (isDateObject(value)) {
+        return new Date(value).getTime()
+      }
+
+      return value;
+    default:
+      return String(value);
+  }
+}
+
+/**
+ * Check if value is a Date object
+ * 
+ * @param {any} value 
+ */
+export const isDateObject = (value: any): boolean => Object.prototype.toString.call(value) === '[object Date]';
```

Comment:
Do we need to export this? seems like an inside thingy! 

Also could this be simplified to ? 
```suggestion
const isDateObject = (value: any): boolean => value instanceof Date';
```

Reply:
https://stackoverflow.com/a/643827

`new Date("some random text")` is an instance of Date, so this could lead to false positives. This way we will know for certain that this is a valid date object.

Reply:
No need to export it (I was at one time using it from another file).

Reply:
> https://stackoverflow.com/a/643827
> 
> `new Date("some random text")` is an instance of Date, so this could lead to false positives. This way we will know for certain that this is a valid date object.

Yes, the Date constructor is a weird one but I do not think that this can happen though as some scalars library will perform the validation for us. e.g https://github.com/Urigo/graphql-scalars/blob/dc0c2bdfdbf6742b79c0a223024fe2fee1329457/src/scalars/iso-date/DateTime.ts#L36

Reply:
Correct, but as you mentioned this filter can be used anywhere in theory, and could be used without going through a GraphQL scalar, avoiding validation. Would it be better to cater both cases?

By the way, this is used externally actually: https://github.com/aerogear/graphback/blob/fc8814c0156b70003156bb6838762bf351fed126/packages/graphback-core/src/runtime/subscriptionFilter.ts#L62

Reply:
Okay, we can leave this as it is now.